steps/bootstrap/aws/main.tf

```diff
@@ -14,144 +9,28 @@ provider "aws" {
-}
-
-data "aws_iam_role" "bootstrap_role" {
-  count = "${var.tectonic_aws_master_iam_role_name == "" ? 0 : 1}"
-  name  = "${var.tectonic_aws_master_iam_role_name}"
-}
-
-resource "aws_iam_role" "bootstrap_role" {
-  count = "${var.tectonic_aws_master_iam_role_name == "" ? 1 : 0}"
-  name  = "${var.tectonic_cluster_name}-bootstrap-role"
-  path  = "/"
-
-  assume_role_policy = <<EOF
-{
-    "Version": "2012-10-17",
-    "Statement": [
-        {
-            "Action": "sts:AssumeRole",
-            "Principal": {
-                "Service": "ec2.amazonaws.com"
-            },
-            "Effect": "Allow",
-            "Sid": ""
-        }
-    ]
-}
-EOF
-}
-
-resource "aws_iam_role_policy" "bootstrap_policy" {
-  count = "${var.tectonic_aws_master_iam_role_name == "" ? 1 : 0}"
-  name  = "${var.tectonic_cluster_name}-bootstrap-policy"
-  role  = "${aws_iam_role.bootstrap_role.id}"
-
-  policy = <<EOF
-{
-  "Version": "2012-10-17",
-  "Statement": [
-    {
-      "Effect": "Allow",
-      "Action": "ec2:Describe*",
-      "Resource": "*"
-    },
-    {
-      "Effect": "Allow",
-      "Action": "ec2:AttachVolume",
-      "Resource": "*"
-    },
-    {
-      "Effect": "Allow",
-      "Action": "ec2:DetachVolume",
-      "Resource": "*"
-    },
-    {
-      "Action" : [
-        "s3:GetObject"
-      ],
-      "Resource": "arn:aws:s3:::*",
-      "Effect": "Allow"
-    }
-  ]
-}
-EOF
-}
+module "bootstrap" {
+  source = "../../../modules/aws/bootstrap"
 
-resource "aws_instance" "bootstrap" {
-  ami = "${coalesce(var.tectonic_aws_ec2_ami_override, module.ami.id)}"
-
-  iam_instance_profile        = "${aws_iam_instance_profile.bootstrap.name}"
-  instance_type               = "${var.tectonic_aws_master_ec2_type}"
+  ami                         = "${coalesce(var.tectonic_aws_ec2_ami_override, module.ami.id)}"
+  associate_public_ip_address = "${var.tectonic_aws_endpoints != "private"}"
+  bucket                      = "${local.s3_bucket}"
+  cluster_name                = "${var.tectonic_cluster_name}"
+  elbs                        = "${local.aws_lbs}"
+  iam_role                    = "${var.tectonic_aws_master_iam_role_name}"
+  ignition                    = "${local.ignition_bootstrap}"
   subnet_id                   = "${local.subnet_ids[0]}"
-  user_data                   = "${data.ignition_config.bootstrap_redirect.rendered}"
   vpc_security_group_ids      = ["${concat(var.tectonic_aws_master_extra_sg_ids, list(local.sg_id))}"]
-  associate_public_ip_address = "${local.public_endpoints}"
-
-  lifecycle {
-    # Ignore changes in the AMI which force recreation of the resource. This
-    # avoids accidental deletion of nodes whenever a new OS release comes out.
-    ignore_changes = ["ami"]
-  }
 
   tags = "${merge(map(
       "Name", "${var.tectonic_cluster_name}-bootstrap",
-      "kubernetes.io/cluster/${var.tectonic_cluster_name}", "owned",
```

Comment:
why drop this tag?

Reply:
> why drop this tag?

Because we're eventually going to tear down the bootstrap stuff as part of installation, so we aren't leaving it around for Kubernetes to own.